release/ray_release/test_automation/state_machine.py

```diff
@@ -1,10 +1,18 @@
+import boto3
+from github import Github
+
 from ray_release.test import (
     Test,
     TestState,
 )
 
+RAY_REPO = "ray-project/ray"
+AWS_SECRET_GITHUB = "ray_ci_github_token"
+
 
 class TestStateMachine:
+    ray_repo = None
```

Comment:
this is a class attribute, which is shared for all the instances of this class.

which means that when you have multiple of `TestStateMachine`, line 33's check and lazy set can be a racing condition where the secret gets read twice.

so it is a pattern that is a bit confusing.

Reply:
if you want to stub/fake `ray_repo`, you can send in an optional arg on construction, or maybe just alternate `self.ray_repo` directly after init.